From 7d8c06d5e76c90953bbda22a31256b05ceef3798 Mon Sep 17 00:00:00 2001
From: Matt Casey <mrcasey@google.com>
Date: Tue, 17 Jun 2025 19:30:30 +0000
Subject: [PATCH] Launch image editor as sharesheet-launching user (flag off)

ag/34038177 but for the cases where use_preferred_image_editor is off or
does not exist.

Bug: 407991863
Test: Manual repro of bug and verification of fix.
Test: atest IntentResolver-tests-*
Flag: EXEMPT bugfix
Change-Id: Idf2d76615f4396f09e7de33becc87afaea782d41
Merged-In: Idf2d76615f4396f09e7de33becc87afaea782d41
(cherry picked from commit 29da3d663edd566f0b5c4dce434f40aa7e3877de)
---
 .../android/intentresolver/ChooserActionFactory.java   | 10 +++++-----
 .../com/android/intentresolver/ChooserActivity.java    |  8 ++++----
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/java/src/com/android/intentresolver/ChooserActionFactory.java b/java/src/com/android/intentresolver/ChooserActionFactory.java
index 21ca3b73..160901b5 100644
--- a/java/src/com/android/intentresolver/ChooserActionFactory.java
+++ b/java/src/com/android/intentresolver/ChooserActionFactory.java
@@ -68,16 +68,16 @@ public final class ChooserActionFactory implements ChooserContentPreviewUi.Actio
          * Request an activity launch for the provided target. Implementations may choose to exit
          * the current activity when the target is launched.
          */
-        void safelyStartActivityAsPersonalProfileUser(TargetInfo info);
+        void safelyStartActivityAsLaunchingUser(TargetInfo info);
 
         /**
          * Request an activity launch for the provided target, optionally employing the specified
          * shared element transition. Implementations may choose to exit the current activity when
          * the target is launched.
          */
-        default void safelyStartActivityAsPersonalProfileUserWithSharedElementTransition(
+        default void safelyStartActivityAsLaunchingUserWithSharedElementTransition(
                 TargetInfo info, View sharedElement, String sharedElementName) {
-            safelyStartActivityAsPersonalProfileUser(info);
+            safelyStartActivityAsLaunchingUser(info);
         }
     }
 
@@ -350,9 +350,9 @@ public final class ChooserActionFactory implements ChooserContentPreviewUi.Actio
             } catch (Exception e) { /* ignore */ }
             // Action bar is user-independent; always start as primary.
             if (firstImageView == null || !isFullyVisible(firstImageView)) {
-                activityStarter.safelyStartActivityAsPersonalProfileUser(editSharingTarget);
+                activityStarter.safelyStartActivityAsLaunchingUser(editSharingTarget);
             } else {
-                activityStarter.safelyStartActivityAsPersonalProfileUserWithSharedElementTransition(
+                activityStarter.safelyStartActivityAsLaunchingUserWithSharedElementTransition(
                         editSharingTarget, firstImageView, IMAGE_EDITOR_SHARED_ELEMENT);
             }
         };
diff --git a/java/src/com/android/intentresolver/ChooserActivity.java b/java/src/com/android/intentresolver/ChooserActivity.java
index 6df9e948..f92f557f 100644
--- a/java/src/com/android/intentresolver/ChooserActivity.java
+++ b/java/src/com/android/intentresolver/ChooserActivity.java
@@ -2236,10 +2236,10 @@ public class ChooserActivity extends Hilt_ChooserActivity implements
                 this::getFirstVisibleImgPreviewView,
                 new ChooserActionFactory.ActionActivityStarter() {
                     @Override
-                    public void safelyStartActivityAsPersonalProfileUser(TargetInfo targetInfo) {
+                    public void safelyStartActivityAsLaunchingUser(TargetInfo targetInfo) {
                         safelyStartActivityAsUser(
                                 targetInfo,
-                                mProfiles.getPersonalHandle()
+                                mUserInteractor.getLaunchedAs()
                         );
                         Log.d(TAG, "safelyStartActivityAsPersonalProfileUser("
                                 + targetInfo + "): finishing!");
@@ -2247,13 +2247,13 @@ public class ChooserActivity extends Hilt_ChooserActivity implements
                     }
 
                     @Override
-                    public void safelyStartActivityAsPersonalProfileUserWithSharedElementTransition(
+                    public void safelyStartActivityAsLaunchingUserWithSharedElementTransition(
                             TargetInfo targetInfo, View sharedElement, String sharedElementName) {
                         ActivityOptions options = ActivityOptions.makeSceneTransitionAnimation(
                                 ChooserActivity.this, sharedElement, sharedElementName);
                         safelyStartActivityAsUser(
                                 targetInfo,
-                                mProfiles.getPersonalHandle(),
+                                mUserInteractor.getLaunchedAs(),
                                 options.toBundle());
                         // Can't finish right away because the shared element transition may not
                         // be ready to start.
-- 
2.34.1

